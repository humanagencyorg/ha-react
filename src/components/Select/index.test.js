import React from 'react';
import renderer from 'react-test-renderer';

import { Select } from './index';
import { mount } from 'enzyme';

describe('Select component', () => {
  describe('without error', () => {
    it('matches snapshot', () => {
      const options = [{ value: 'value_1', name: 'name_1' }];
      const props = {
        options,
        name: 'name',
        label: 'label',
      };

      const wrapper = renderer.create(<Select {...props} />);

      expect(wrapper.toJSON()).toMatchSnapshot();
    });
  });

  describe('with error', () => {
    it('matches snapshot', () => {
      const options = [{ value: 'value_1', name: 'name_1' }];
      const props = {
        options,
        name: 'name',
        label: 'label',
        error: true,
        helperText: 'error',
      };

      const wrapper = renderer.create(<Select {...props} />);

      expect(wrapper.toJSON()).toMatchSnapshot();
    });

    it('shows error message', () => {
      const options = [{ value: 'value_1', name: 'name_1' }];
      const errorMessage = 'error';
      const props = {
        options,
        error: true,
        helperText: errorMessage,
      };

      const wrapper = mount(<Select {...props} />);
      const errorWrapper = wrapper.find('p.Mui-error');

      expect(errorWrapper.text()).toBe(errorMessage);
    });
  });

  it('sets default value from props', () => {
    const options = [{ value: 'value_1', name: 'name_1' }];
    const defaultValue = 'value_1';
    const props = {
      options,
      defaultValue,
    };

    const wrapper = mount(<Select {...props} />);
    const select = wrapper.find('select');

    expect(select.prop('value')).toBe(defaultValue);
  });

  it('calls #onFocus from props on field focus', () => {
    const onFocus = jest.fn();
    const options = [{ value: 'value_1', name: 'name_1' }];
    const props = {
      options,
      onFocus,
    };

    const wrapper = mount(<Select {...props} />);
    const select = wrapper.find('select');

    select.simulate('focus');

    expect(onFocus).toHaveBeenCalled();
  });
});
