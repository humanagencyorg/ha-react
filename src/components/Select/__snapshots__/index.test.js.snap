// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select component with error matches snapshot 1`] = `
<div
  className="MuiFormControl-root makeStyles-root-1"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root makeStyles-label-2 MuiInputLabel-formControl MuiInputLabel-animated Mui-error Mui-error"
    data-shrink={false}
    htmlFor="data_response_country"
  >
    label
  </label>
  <div
    className="MuiInputBase-root Mui-error MuiInputBase-formControl"
    onClick={[Function]}
  >
    <select
      aria-invalid={true}
      className="MuiNativeSelect-root MuiNativeSelect-select MuiInputBase-input WithStyles(ForwardRef(InputBase))-input-6"
      disabled={false}
      id="data_response_country"
      name="name"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      value=""
    >
      <option
        value="value_1"
      >
        name_1
      </option>
    </select>
    <svg
      className="MuiNativeSelect-icon makeStyles-icon-4"
      fill="none"
      height={24}
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      viewBox="0 0 24 24"
      width={24}
      xmlns="http://www.w3.org/2000/svg"
    >
      <polyline
        points="6 9 12 15 18 9"
      />
    </svg>
  </div>
  <p
    className="MuiFormHelperText-root makeStyles-errorMessage-3 Mui-error"
  >
    error
  </p>
</div>
`;

exports[`Select component without error matches snapshot 1`] = `
<div
  className="MuiFormControl-root makeStyles-root-1"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root makeStyles-label-2 MuiInputLabel-formControl MuiInputLabel-animated"
    data-shrink={false}
    htmlFor="data_response_country"
  >
    label
  </label>
  <div
    className="MuiInputBase-root MuiInputBase-formControl"
    onClick={[Function]}
  >
    <select
      aria-invalid={false}
      className="MuiNativeSelect-root MuiNativeSelect-select MuiInputBase-input WithStyles(ForwardRef(InputBase))-input-6"
      disabled={false}
      id="data_response_country"
      name="name"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      value=""
    >
      <option
        value="value_1"
      >
        name_1
      </option>
    </select>
    <svg
      className="MuiNativeSelect-icon makeStyles-icon-4"
      fill="none"
      height={24}
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      viewBox="0 0 24 24"
      width={24}
      xmlns="http://www.w3.org/2000/svg"
    >
      <polyline
        points="6 9 12 15 18 9"
      />
    </svg>
  </div>
</div>
`;
